### DISCUSSION TITLE
mypy: error: Untyped decorator makes function ... untyped - caused by hookimpl

### DISCUSSION THREAD

**Discussion by ssbarnea** [pytest-dev/pytest#11513] [Q&A]
```
src/pytest_plus/__init__.py:33: error: Untyped decorator makes function "pytest_collection_modifyitems" untyped  [misc]
```

```python
@pytest.hookimpl(tryfirst=True)
def pytest_collection_modifyitems(items: list[pytest.Item]) -> None:
  ...
```

To be honest, I have no idea what I can do to make mypy happy with this error. Not sure how to add types to the decorator itself.

Reference: https://github.com/pytest-dev/pytest-plus/actions/runs/6549768159/job/17787433353?pr=20

**Comment by RonnyPfannschmidt** [comment]
This possibly needs a pluggy update, pluggy exported types relatively recently
